### ISSUE TITLE
Clippy doesn't warn on `self` parameter for lints `pass-by-value-size-limit` and `trivial-copy-size-limit`

### ISSUE THREAD

**Issue by x4exr** [rust-lang/rust-clippy#13212]
### Summary

Both lints work correctly on parameters that aren't self for methods. 

### Lint Name

pass-by-value-size-limit, trivial-copy-size-limit

### Reproducer

I tried this code:

```rust
// pass-by-value-size-limit
struct Ins {
    inner: [u8; 128]
}

impl Ins {
    pub fn some_thing(self) {
        // .. Expected error self should be passed as reference more efficiently.
    }
}
```

```rs
// trivial-copy-size-limit
struct TestO {
    inner: u8
}

impl TestO {
    pub fn some_thing(&self) {
        // Expected error &self should be passed as copy more efficiently.
    } 
}
```

I expected to see this happen:

Instead, this happened:

### Version

```text
rustc 1.81.0-nightly (bcf94dec5 2024-06-23)
binary: rustc
commit-hash: bcf94dec5ba6838e435902120c0384c360126a26
commit-date: 2024-06-23
host: x86_64-pc-windows-msvc
release: 1.81.0-nightly
LLVM version: 18.1.7
```

**Comment by x4exr** [comment]
With furthur testing, clippy fails to do this check on arrays too. [Type; N] or [Type; GENERIC_CONST]

**Comment by x4exr** [comment]
![image](https://github.com/user-attachments/assets/7f16cd1d-cc24-4bef-b16b-cb5cc8566f05)
![image](https://github.com/user-attachments/assets/7d4c1999-ca2b-4278-99f6-d4e67a8640a5)

For some odd reason.
